Tests/Services/ParkingSlotServiceTests.cs

```diff
@@ -175,5 +177,148 @@ public void GivenNumberAndIdOfExistingSlotAndZoneId_WhenSlotWithThisNumberExists
+                new()
+                {
+                        IsAvailableForBooking = true,
+                        ParkingZoneId = _testZoneId,
+                        Reservations = new List<Reservation>()
+                        {
+                            new()
+                            {
+                                Id = _testReservationId,
+                                StartTime = new DateTime(2024, 1, 27, 21, 00, 00),
+                                Duration = 2
+                            }
+                        }
+                },
+                new()
+                {
+                        IsAvailableForBooking = true,
+                        ParkingZoneId = _testZoneId,
+                        Reservations = new List<Reservation>()
+                        {
+                            new()
+                            {
+                                Id = _testReservationId,
+                                StartTime = new DateTime(2024, 1, 27, 13, 00, 00),
+                                Duration = 5
+                            }
+                        }
+                }
+            };
+
+            var bookedSlots = new List<ParkingSlot>()
+            {
+                new()
+                {
+                        IsAvailableForBooking = true,
+                        ParkingZoneId = _testZoneId,
+                        Reservations = new List<Reservation>()
+                        {
+                            new()
+                            {
+                                Id = _testReservationId,
+                                StartTime = new DateTime(2024, 1, 27, 18, 00, 00),
+                                Duration = 3
+                            }
+                        }
+                },
+                new()
+                {
+                        IsAvailableForBooking = false,
+                        ParkingZoneId = _testZoneId,
+                        Reservations = new List<Reservation>()
+                        {
+                            new()
+                            {
+                                Id = _testReservationId,
+                                StartTime = new DateTime(2024, 1, 27, 19, 00, 00),
+                                Duration = 1
+                            }
+                        }
+                }
+            };
+
+            var all_slots = freeSlots.Concat(bookedSlots);
+
+            mockSlotRepository
+                .Setup(repo => repo.GetAllWithReservations())
+                .Returns(all_slots);
+
+            //Act
+            var result = service.GetFreeByZoneIdAndTimePeriod(_testZoneId, testStartTime, testDuration);
+
+            //Assert
+            Assert.Equal(JsonSerializer.Serialize(freeSlots), JsonSerializer.Serialize(result));
+            mockSlotRepository.Verify(repo => repo.GetAllWithReservations(), Times.Once);
+        }
+        #endregion
+
+        #region IsSlotFree
```

Comment:
Investigate [this test](https://github.com/DilshodMQ/Parking-zone/pull/15/files#diff-d857dec5ab070ea7dada2c6fcc77de3aab98e66aaf31986f0d48c99b36acf468R284) and make appropriate changes.